### PR TITLE
Search payment

### PR DESCRIPTION
**Pull request by Developers-codz**
Features added - Search products, Test mode Payment integration
Bug fixing of cart and wishlist

### CODE REVIEW

**Review by vishalg8454 — CHANGES_REQUESTED** [review]
Bugs i noticed

- Negative and zero values in cart.

image

- Same item multiple times in wish-list

image

**Comment by Developers-codz** [comment]
> Bugs i noticed
> 
> * Negative and zero values in cart.
> 
> image
> 
> * Same item multiple times in wish-list
> 
> image

Thank you for pointing those bugs out but those are coming because you are continuously pressing the button and making the function run multiple times. I will fix them soon :)

**Review by vishal804 — APPROVED** [review]

**Comment by vishal804** [comment]
Code Looks Good and Clean

**Comment by Developers-codz** [comment]
> Bugs i noticed
> 
> * Negative and zero values in cart.
> 
> image
> 
> * Same item multiple times in wish-list
> 
> image

Its done can you check it now and approve it  :)

**Review by vishalg8454 — APPROVED** [review]
Good UI, all edge cases are properly handled now. Approval for merge!

File: .env [hunk 1]
```diff
@@ -1 +1,2 @@
-REACT_APP_JWT_SECRET=3451278
\ No newline at end of file
+REACT_APP_JWT_SECRET=3451278
```
Comment (MarkVed17) [inline .env:1]: Never let your secret keys public.
Add your ‘env’ to .gitignore

File: src/functions/debounce.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export const debounce = (func,delay)=>{
```
Comment (MarkVed17) [inline src/functions/debounce.js:1]: Oh, debounce implementation. Learnt something new here.

**Review by MarkVed17 — APPROVED** [review]
Never forget to hide your secret keys. You’re good to proceed.

**Review by simran61 — APPROVED** [review]
Looks good

**Comment by Developers-codz** [comment]
Thank you for all the suggestions and review. Merging it now

**Pull request merged**